qa-tests-backend/src/test/groovy/VulnMgmtSACTest.groovy

```diff
@@ -130,13 +313,120 @@ class VulnMgmtSACTest extends BaseSpecification {
         cleanup:
         "Cleanup"
         BaseService.useBasicAuth()
+    }
+
+    @Retry(count = 0)
+    @Unroll
+    def "Verify role based scoping on vuln mgmt: image-role Image:*"() {
+        when:
+        "Get Node CVEs and components"
+        BaseService.useBasicAuth()
+        def gqlService = new GraphQLService()
+        def baseQuery = "Image:*"
+        def cveQuery = getImageCVEQuery()
+        def componentQuery = getImageComponentQuery()
+        def baseVulnCallResult = gqlService.Call(cveQuery, [query: baseQuery])
+        assert baseVulnCallResult.hasNoErrors()
+        def baseComponentCallResult = gqlService.Call(componentQuery, [query: baseQuery])
+        assert baseComponentCallResult.hasNoErrors()
+
+        and:
+        gqlService = new GraphQLService(getToken(IMAGE_ROLE, IMAGE_ROLE))
+        def vulnCallResult = gqlService.Call(cveQuery, [query: ""])
+        assert vulnCallResult.hasNoErrors()
+        def componentCallResult = gqlService.Call(componentQuery, [query: ""])
+        assert componentCallResult.hasNoErrors()
+        def baseSortedVulns = extractCVEsAndSort(baseVulnCallResult.value)
+        def sortedVulns = extractCVEsAndSort(vulnCallResult.value)
+        if ( baseSortedVulns != sortedVulns ) {
```

Comment:
If we have sets it will be easier to write (aka need less indents)
```groovy
        baseSortedVulns-sortedVulns.each { String v -> 
            log.info("Item found in baseVulnCallResult but not in vulnCallResults: " + v)
            for ( img in getImagesWithCVE(v) ) {
                log.debug "Image ${img} has vulnerability ${v}"
            }
        }
```